### PR TITLE
feat(lint): add useTanStackQueryExhaustiveDeps rule

### PR DESCRIPTION
**Pull request by washingtonsoares**
## Summary

  Implements `useTanStackQueryExhaustiveDeps` rule to enforce explicit dependency listing in TanStack Query hooks
  (`useQuery` and `useInfiniteQuery`), preventing stale closures and ensuring proper query invalidation.

  This addresses a common source of bugs in React applications using TanStack Query where queries don't re-run when
  their dependencies change, similar to the React `useEffect` exhaustive deps rule.

  This is a beta implementation that covers core use cases for dependency tracking in query keys.

  ## Test Plan

  - Unit tests covering basic dependency detection scenarios
  - Tests for object destructuring and member expression patterns
  - Edge case validation for nested callbacks and conditional logic
  - Snapshot tests for various TanStack Query hook configurations
  - Manual testing with real TanStack Query codebases

  ## Docs

  Documentation is included in the rule implementation with comprehensive examples showing invalid/valid patterns and
  detailed explanations of the rule's behavior.

  **Key Future Improvements:**
  1. **Enhanced scope analysis** - Better handling of complex nested scopes and conditional dependencies
  2. **Framework-specific patterns** - Support for custom hook patterns and framework-specific query compositions
  3. **Performance optimization** - Reduce false positives for stable references and memoized values

**Note:** Template literal patterns in queryKey are currently not supported by this rule. Support for template literal dependency detection will be added in a future update.
Example:

```js
// Dep included via template literal in queryKey (should be valid)
function Component9({ userId }) {
	const query = useQuery({
		queryKey: ["users", `u:${userId}`],
		queryFn: () => fetchUser(userId),
	});
}
```

### CODE REVIEW

File: .changeset/add-tanstack-query-exhaustive-deps-rule.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+"@biomejs/biome": minor
```
Comment (siketyan) [inline .changeset/add-tanstack-query-exhaustive-deps-rule.md:2]: Nursury rules can be added in a patch release.
```suggestion
"@biomejs/biome": patch
```

File: crates/biome_js_analyze/tests/specs/nursery/useTanStackQueryExhaustiveDeps/import_detection.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+// TanStack: Legacy package name
+import { useQuery } from 'react-query';
+
+function ValidComponent4({ userId }) {
+  const query = useQuery({
+    queryKey: ['users'],
+    queryFn: () => fetchUser(userId) // Should report missing userId
+  });
+}
```
Comment (siketyan) [inline crates/biome_js_analyze/tests/specs/nursery/useTanStackQueryExhaustiveDeps/import_detection.js:39]: TanStack Query <= v4 had the different signature of `useQuery`. It would be great if the rule also support that cases.

https://tanstack.com/query/v5/docs/framework/react/guides/migrating-to-v5#supports-a-single-signature-one-object

File: crates/biome_js_analyze/src/lint/nursery/use_tan_stack_query_exhaustive_deps.rs [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+    ///     queryFn: () => fetchUser(userId)
+    ///   });
+    /// }
+    /// ```
+    ///
+    pub UseTanStackQueryExhaustiveDeps {
+        version: "next",
+        name: "useTanStackQueryExhaustiveDeps",
+        language: "js",
+        sources: &[RuleSource::EslintReactHooks("exhaustive-deps").same()],
```
Comment (siketyan) [inline crates/biome_js_analyze/src/lint/nursery/use_tan_stack_query_exhaustive_deps.rs:53]: This is not true. Please add `@tanstack/eslint-plugin-query` to the `RuleSource` enum and use it here instead.

File: crates/biome_js_analyze/src/lint/nursery/use_tan_stack_query_exhaustive_deps.rs [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+        )
+    }
+}
+
+/// List of valid TanStack Query package names
+const TANSTACK_QUERY_PACKAGES: &[&str] = &[
+    "@tanstack/react-query",
+    "@tanstack/react-query/experimental",
+    "react-query", // Legacy package name
+];
```
Comment (dyc3) [inline crates/biome_js_analyze/src/lint/nursery/use_tan_stack_query_exhaustive_deps.rs:126]: This should include the package names for other frameworks, like vue/svelte

File: crates/biome_js_analyze/src/lint/nursery/use_tan_stack_query_exhaustive_deps.rs [hunk 5]
```diff
@@ -0,0 +1,524 @@
...
+        _ => false,
+    };
+
+    if !is_target_hook {
+        return false;
+    }
+
+    // If there's a binding, verify it's imported from TanStack Query
+    if let Some(binding) = binding_to_check {
+        is_tanstack_query_import(&binding, &hook_name)
```
Comment (adamjedlicka) [inline crates/biome_js_analyze/src/lint/nursery/use_tan_stack_query_exhaustive_deps.rs:178]: In our codebase we use alias for the import (so the `useQuery` is not imported directly from the Tanstack Query package). It would be useful if this could be configurable (or turned off) somehow.

**Comment by ematipico** [comment]
@washingtonsoares are you still interested in this PR?

File: crates/biome_js_analyze/src/lint/nursery/use_tan_stack_query_exhaustive_deps.rs (hunk 3 above)
Comment (Netail) [reply crates/biome_js_analyze/src/lint/nursery/use_tan_stack_query_exhaustive_deps.rs:53]: Best to first pull in main (due to some order changes) before adding this RuleSource to `crates/biome_analyze/src/rule.rs`
